src/marqo/client.py

```diff
@@ -199,12 +200,34 @@ def get_cpu_info(self):
         return self.http.get(path="device/cpu")
 
     def _marqo_minimum_supported_version_check(self):
-        if self.url not in marqo_url_and_version_cache:
-            marqo_url_and_version_cache[self.url] = self.get_marqo()["version"]
-        marqo_version = marqo_url_and_version_cache[self.url]
-        if versioning_helpers.parse(marqo_version) < versioning_helpers.parse(minimum_supported_marqo_version()):
-            mq_logger.warning(f"Your Marqo Python client requires a minimum Marqo version of "
-                              f"{minimum_supported_marqo_version()} to function properly, while your Marqo version is {marqo_version}. "
-                              f"Please upgrade your Marqo instance to avoid potential errors. "
-                              f"If you have already upgraded your Marqo instance but still having this warning, please import you Marqo Python client again and retry.")
+        min_ver = minimum_supported_marqo_version()
+
+        # Skip the check if the url is previously labelled as "_skipped"
+        if self.url in marqo_url_and_version_cache and marqo_url_and_version_cache[self.url] == "_skipped":
+            return
+
+        # Skip the check for Marqo CloudV2 APIs right now
+        skip_version_check_url = ["https://api.marqo.ai", "https://cloud.marqo.ai"]
+        if self.url in skip_version_check_url:
+            marqo_url_and_version_cache[self.url] = "_skipped"
+            return
+
+        # Do version check
+        try:
+            if self.url not in marqo_url_and_version_cache:
+                marqo_url_and_version_cache[self.url] = self.get_marqo()["version"]
+            marqo_version = marqo_url_and_version_cache[self.url]
+            if versioning_helpers.parse(marqo_version) < versioning_helpers.parse(min_ver):
+                mq_logger.warning(f"Your Marqo Python client requires a minimum Marqo version of "
+                                  f"{minimum_supported_marqo_version()} to function properly, while your Marqo version is {marqo_version}. "
+                                  f"Please upgrade your Marqo instance to avoid potential errors. "
+                                  f"If you have already upgraded your Marqo instance but still having this warning, please import you Marqo Python client again and retry.")
+        except JSONDecodeError as e:
```

Comment:
In what situation do we see a `JSONDecodeError` happening in the version check?

Reply:
When it reaches the cloud v2 API and it is not in the skip list. One example is “https://api.marqo.ai/api/”